build_tools/aws-sdk-code-generator/templates/root_resource_class.mustache

```diff
@@ -2,7 +2,15 @@
 {{generated_src_warning}}
 {{/generated_src_warning}}
 module {{module_name}}
-  class Resource
+  # This class provides a resource oriented interface for {{service_name}}.
+  # To create a resource object:
+  #     resource = {{module_name}}::Resource.new(region: 'us-west-2')
+  # You can supply a client object with custom configuration that will be used for all resource operations.
+  # If you do not pass :client, a default client will be constructed.
+  #     client = {{module_name}}::Client.new(region: 'us-west-2')
+  #     resource = {{module_name}}::Resource.new(client: client)
+  #
+class Resource
```

Comment:
Indenting is wrong here. Check the generated files.

Reply:
I believe the indenting is correct - it needs to be 5 spaces (the standard 1 before the line + 4 for the code formatting) to get yard to format this as code.  This generates:

```html
<p>You can supply a client object with custom configuration that will be used for all resource operations. If you do not pass :client, a default client will be constructed.</p>

<pre class="code ruby"><code class="ruby"><span class='id identifier rubyid_client'>client</span> <span class='op'>=</span> <span class='const'><span class='object_link'><a href="/docs/Aws" title="Aws (module)">Aws</a></span></span><span class='op'>::</span><span class='const'><span class='object_link'><a href="/docs/Aws/S3" title="Aws::S3 (module)">S3</a></span></span><span class='op'>::</span><span class='const'><span class='object_link'><a href="/docs/Aws/S3/Client" title="Aws::S3::Client (class)">Client</a></span></span><span class='period'>.</span><span class='id identifier rubyid_new'><span class='object_link'><a href="/docs/Aws/S3/Client#initialize-instance_method" title="Aws::S3::Client#initialize (method)">new</a></span></span><span class='lparen'>(</span><span class='label'>region:</span> <span class='tstring'><span class='tstring_beg'>&#39;</span><span class='tstring_content'>us-west-2</span><span class='tstring_end'>&#39;</span></span><span class='rparen'>)</span>
```